blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/BluefloodServiceStarter.java

```diff
@@ -306,9 +306,10 @@ public void run() {
         final Collection<Integer> shards = Collections.unmodifiableCollection(
                 Util.parseShards(config.getStringProperty(CoreConfig.SHARDS)));
         final String zkCluster = config.getStringProperty(CoreConfig.ZOOKEEPER_CLUSTER);
-        final ScheduleContext rollupContext = "NONE".equals(zkCluster) ?
-                new ScheduleContext(System.currentTimeMillis(), shards) :
-                new ScheduleContext(System.currentTimeMillis(), shards, zkCluster);
+        final boolean rollupMode = config.getBooleanProperty(CoreConfig.ROLLUP_MODE);
+        final ScheduleContext rollupContext = !"NONE".equals(zkCluster) && rollupMode ?
```

Comment:
The logic is the same (the if and else cases are switched). There's just an additional check to make sure Blueflood is in rollup mode.